### PR TITLE
feat: strip embedded fonts during EPUB optimization

### PR DESCRIPTION
**Pull request by hyemin-ht-kang**
Per @serialx's recommendation and @Uri-Tauber's request in [#1757 (comment)](https://github.com/crosspoint-reader/crosspoint-reader/discussions/1757#discussioncomment-17997328), this PR upstreams my commit `ae85f2a5` from serialx's fork. This PR is reopened with the earlier submission's issues fixed and the change hardened by review and device testing.

## Summary

* **What is the goal of this PR?** The browser-side Optimize & Upload pipeline now strips embedded fonts from EPUBs. CrossPoint renders with built-in / SD-card fonts and never loads fonts embedded in a book, so they are dead weight — commonly 100 KB–2 MB, and up to 18.5 MB in the Korean novels I tested. Removing them shrinks books dramatically and reduces on-device unzip/parse cost.
* **What changes are included?** One file, `src/network/html/FilesPage.html` (+139/−4):
  * Font detection by file extension (`ttf/otf/woff/woff2/eot`, case-insensitive) and by OPF manifest media-type (`font/*`, `application/font-*`, `application/x-font-*`, `vnd.ms-opentype`, `vnd.ms-fontobject`) for odd extensions
  * OPF manifest entries for removed fonts are deleted (DOM path with regex fallback, same structure as the existing `fixOPF` repairs)
  * `@font-face` rules stripped from CSS files **and** inline XHTML `<style>` blocks, including `data:` URI fonts embedded directly in CSS
  * Font-obfuscation entries removed from `META-INF/encryption.xml`; the file is dropped entirely when nothing meaningful remains
  * Conversion log reports each removal (`[REMOVE]`/`[FIX]` lines) and a total: fonts removed, `@font-face` rules removed, bytes saved across all strip sites

## Scope Check

- [x] I have read SCOPE.md and ROADMAP.md.
- [x] This PR is **not** a new built-in theme (themes are temporarily closed pending the move to SD-loaded themes).
- [x] This PR is **not** a new external network connector (sync engine, cloud storage, remote file access, etc.).
- [x] This PR is **not** an interactive app, writing tool, RSS/news/browser, media playback, or PDF feature.
- [x] The stock firmware does not already handle this well, **and** no other popular CrossPoint fork already does (or, if one does, I explain why CrossPoint still needs it below).
- [x] If this PR touches `freeink-sdk/`, `lib/hal/`, the bootloader, OTA, or recovery code, I have coordinated with the relevant maintainer. *(N/A — this PR touches none of those areas.)*

Related: the CrossPoint Calibre plugin already strips fonts off-device, which validates the approach — this PR brings the same capability to users who upload through the built-in web optimizer without Calibre. Complementary to #1825 (font *extraction/conversion* for the web optimizer): stripping is the simple default; extraction can layer on top later.

## Additional Context

* **Flash impact (measured):** +1,768 bytes (5,697,185 → 5,698,953; +0.031%) on the `default` env. The page is minified and gzip-compressed at embed time, so the ~120 lines of JS cost little.
* **EPUB savings (measured, on device):** two Korean novels tested — 7.3 MB → 305.6 KB (2 fonts, 16.7 MB uncompressed removed) and 13.1 MB → 459.0 KB, −96.6% (4 fonts, 18.5 MB removed). Both open and render correctly on an X3 with built-in fonts.
* **Validation:** optimized output passes epubcheck 5.3.0 with zero errors on 8 test books — IDPF samples (including `wasteland-otf-obf` and `wasteland-woff-obf` with obfuscated fonts, where `encryption.xml` is correctly removed), Project Gutenberg #1342, and a synthetic book with `@font-face` inside an inline `<style>` block. No-font EPUBs pass through untouched.
* **Review feedback from the earlier #3085 run is addressed:** inline `<style>` blocks are now stripped, and the bytes-saved report covers CSS/`data:`-URI and `encryption.xml` savings, not just deleted font files.
* **Known limitations (deliberate):** a `}` inside a CSS comment inside an `@font-face` block leaves inert residue (pathological input; the regex structurally cannot eat adjacent rules — its failure mode is under-matching only); the regex fallback for malformed OPFs does not match percent-encoded hrefs (consistent with the existing fallback behavior for split images).
* **Areas to focus on:** the `@font-face` regex and the `encryption.xml` handling are the parts with edge-case surface; both have the test evidence above.

---

### AI Usage

Did you use AI tools to help write this code? _**YES**_ — the implementation and tests were developed with AI assistance under my direction and review; I take full responsibility for the change, and I tested it on real hardware.

### CODE REVIEW

**Comment by hyemin-ht-kang** [comment]
I am sorry it was an accident. I will create a new proper one when it's ready.

File: src/network/html/FilesPage.html [hunk 1]
```diff
@@ -5256,25 +5342,51 @@ <h3>🖼️ Preview</h3>
...
     if (operationCancelled) throw new Error('Cancelled by user');
     if (fileObj.dir || path === 'mimetype') continue;
     const low = path.toLowerCase();
     if (low.match(/\.(png|gif|webp|bmp|jpg|jpeg)$/) || low.match(/\.(xhtml|html|htm)$/) || low.endsWith('.opf')) continue;
 
+    if (fontPaths.has(path)) {
+      const fontData = await fileObj.async('arraybuffer');
+      removedFontCount++;
+      removedFontBytes += fontData.byteLength;
+      log(`${escapeHtml(path.split('/').pop())} <span class="log-detail">(${formatBytes(fontData.byteLength)})</span>`, 'success', 'FONT');
```
[the remark this answers is not in the document]
Comment (hyemin-ht-kang) [reply src/network/html/FilesPage.html]: Both are addressed in the rewritten commit: inline <style> @font-face rules are now stripped, and the savings report now includes byte deltas from CSS and encryption.xml changes alongside removed font files.

**Review by Uri-Tauber — APPROVED** [review]
LGTM

**Comment by Uri-Tauber** [comment]
@hyemin-ht-kang can you address coderabbit first suggestion?

File: src/network/html/FilesPage.html [hunk 2]
```diff
@@ -4124,12 +4124,79 @@ <h3>🖼️ Preview</h3>
...
+  }
+  return fontPaths;
+}
+
+// Strips all @font-face rules from CSS. Rules can also embed fonts directly as
+// multi-hundred-KB data: URIs, so this is a size win even without font files.
+function stripFontFaceRules(css) {
+  let count = 0;
+  const stripped = css.replace(/@font-face\s*\{[^{}]*\}/gi, () => { count++; return ''; });
+  return { css: stripped, count };
```
[the remark this answers is not in the document]
Comment (hyemin-ht-kang) [reply src/network/html/FilesPage.html:4164]: Fixed in 4f7421e3 — the pattern now accepts comments between `@font-face` and the opening brace.

File: src/network/html/FilesPage.html [hunk 3]
```diff
@@ -5268,13 +5370,42 @@ <h3>🖼️ Preview</h3>
     const low = path.toLowerCase();
     if (low.match(/\.(png|gif|webp|bmp|jpg|jpeg)$/) || low.match(/\.(xhtml|html|htm)$/) || low.endsWith('.opf')) continue;
 
+    if (fontPaths.has(path)) {
+      const fontData = await fileObj.async('arraybuffer');
+      removedFontCount++;
+      removedFontBytes += fontData.byteLength;
+      log(`${escapeHtml(path.split('/').pop())} <span class="log-detail">(${formatBytes(fontData.byteLength)})</span>`, 'success', 'REMOVE');
```
[the remark this answers is not in the document]
Comment (hyemin-ht-kang) [reply src/network/html/FilesPage.html:5378]: Fixed in 4f7421e3 — the total now reads "…, N MB of font data" instead of "saved N MB", so it's explicitly content bytes removed; the real archive-size delta remains in the conversion summary (`logSummary`).

File: src/network/html/FilesPage.html [hunk 4]
```diff
@@ -5288,6 +5420,10 @@ <h3>🖼️ Preview</h3>
     out.file(path, data, { compression: 'DEFLATE', compressionOptions: { level: 8 }, createFolders: false });
   }
 
+  if (removedFontCount > 0 || removedFontFaceRules > 0) {
+    log(`Removed ${removedFontCount} embedded font(s) and ${removedFontFaceRules} @font-face rule(s), ${formatBytes(removedFontBytes)} of font data`, '', 'INFO');
```
[the remark this answers is not in the document]
Comment (hyemin-ht-kang) [reply src/network/html/FilesPage.html:5424]: All bytes in this counter are font-related — font files, `@font-face` rule text, and font-obfuscation entries — so "font data" is intentional shorthand for readability; the CSS/encryption deltas it also covers are a fraction of a percent of the total. The exact archive-size saving is reported separately in the conversion summary.

**Comment by hyemin-ht-kang** [comment]
@Uri-Tauber I have addressed the two previous reviews from CodeRabbit, but the third one seems a bit nitpicky. But let me know if you would like me to follow up on it!
